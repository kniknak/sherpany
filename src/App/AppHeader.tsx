import React, { memo } from "react";
import { Link } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Button from "@material-ui/core/Button";

export const AppHeader: React.FC = memo(() => {
    return (
        <AppBar position="static">
            <Toolbar>
                <Button
                    color="inherit"
                    component={Link}
                    to="/"
                >
                    Home
                </Button>
                <Button
                    color="inherit"
                    component={Link}
                    to="/settings"
                >
                    Settings
                </Button>
            </Toolbar>
        </AppBar>
    );
});
