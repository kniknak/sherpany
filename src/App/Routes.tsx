import React, { memo, lazy } from "react";
import { Route } from "react-router-dom";
import { withPreloader } from "../UI/withPreloader";

const AddressBook = lazy(() => import("../AddressBook/AddressBook"));
const SettingsConnected = lazy(() => import("../Settings/Settings"));

export const Routes: React.FC = memo(() => {
    return (
        <>
            <Route
                path="/"
                exact
                component={withPreloader(AddressBook)}
            />
            <Route
                path="/settings"
                component={withPreloader(SettingsConnected)}
            />
        </>
    );
});
