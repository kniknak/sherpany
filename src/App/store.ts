import { configureStore } from "@reduxjs/toolkit";
import { addressBookSlice } from "../AddressBook/redux/addressBookSlice";
import { apiMiddleware } from "redux-api-middleware";
import { TypedUseSelectorHook, useSelector } from "react-redux";
import { enableMapSet } from "immer";

enableMapSet();

export const store = configureStore({
    reducer: { addressBook: addressBookSlice.reducer },
    middleware: [apiMiddleware],
});

export type RootState = ReturnType<typeof store.getState>

export const useTypedSelector: TypedUseSelectorHook<RootState> = useSelector;
