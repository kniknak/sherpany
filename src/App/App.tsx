import React from "react";
import { MuiThemeProvider, Theme } from "@material-ui/core";
import CssBaseline from "@material-ui/core/CssBaseline";
import { Provider } from "react-redux";
import { store } from "./store";
import { Switch, BrowserRouter } from "react-router-dom";
import { AppHeader } from "./AppHeader";
import { Routes } from "./Routes";

export interface AppProps {
    theme: Theme
    store: typeof store
}

export class App extends React.PureComponent<AppProps> {
    render() {
        const { store, theme } = this.props;

        return (
            <Provider store={store}>
                <MuiThemeProvider theme={theme}>
                    <CssBaseline/>
                    <BrowserRouter>
                        <AppHeader/>
                        <Switch>
                            <Routes/>
                        </Switch>
                    </BrowserRouter>
                </MuiThemeProvider>
            </Provider>
        );
    }
}