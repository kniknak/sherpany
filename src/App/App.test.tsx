import React from "react";
import { shallow } from "enzyme";
import { App, AppProps } from "./App";

describe("App", () => {
    it("renders", () => {
        const props: AppProps = {
            theme: {} as any,
            store: {
                subscribe: () => undefined,
                dispatch: () => undefined,
                getState: () => undefined,
            } as any,
        };

        const subject = shallow(<App {...props}/>);

        expect(subject).toMatchSnapshot();
    });
});
