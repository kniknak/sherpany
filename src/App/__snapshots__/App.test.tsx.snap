// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "subscribe": [Function],
    }
  }
>
  <ThemeProvider
    theme={Object {}}
  >
    <WithStyles(CssBaseline) />
    <BrowserRouter>
      <Memo() />
      <Switch>
        <Memo() />
      </Switch>
    </BrowserRouter>
  </ThemeProvider>
</Provider>
`;
